public/app/plugins/datasource/loki/backendResultTransformer.ts

```diff
@@ -0,0 +1,100 @@
+import { DataQueryRequest, DataQueryResponse, DataFrame, isDataFrame, FieldType, QueryResultMeta } from '@grafana/data';
+import { LokiQuery, LokiQueryType } from './types';
+import { makeTableFrames } from './makeTableFrames';
+import { formatQuery, getHighlighterExpressionsFromQuery } from './query_utils';
+
+function isMetricFrame(frame: DataFrame): boolean {
+  return frame.fields.every((field) => field.type === FieldType.time || field.type === FieldType.number);
+}
+
+// returns a new frame, with meta merged with it's original meta
+function setFrameMeta(frame: DataFrame, meta: QueryResultMeta): DataFrame {
+  const { meta: oldMeta, ...rest } = frame;
+  // meta maybe be undefined, we need to handle that
+  const newMeta = oldMeta === undefined ? { ...meta } : { ...oldMeta, ...meta };
```

Comment:
```suggestion
  const newMeta = { ...oldMeta, ...meta };
```

I think that you can do this even if oldMeta is undefined. {...undefined} doesn't produce error https://stackoverflow.com/questions/46957194/javascript-es6-spread-operator-on-undefined